src/sparsezoo/models/zoo.py

```diff
@@ -17,77 +17,53 @@
 """
 
 
-import warnings
-from typing import Dict, List, Tuple, Union
+from typing import Callable, List, Union
 
 from sparsezoo.objects.model import Model
-from sparsezoo.objects.optimization_recipe import (
-    OptimizationRecipe,
-    OptimizationRecipeTypes,
-)
-from sparsezoo.requests import (
-    ModelArgs,
-    RecipeArgs,
-    download_model_get_request,
-    download_recipe_get_request,
-    get_model_get_request,
-    get_recipe_get_request,
-    search_model_get_request,
-    search_recipe_get_request,
-)
+from sparsezoo.objects.optimization_recipe import OptimizationRecipe
+from sparsezoo.requests import ModelArgs
 
 
 __all__ = [
-    "ZOO_STUB_PREFIX",
-    "parse_zoo_stub",
     "Zoo",
 ]
 
 
-# optional prefix for stubs
-ZOO_STUB_PREFIX = "zoo:"
-
-
-def parse_zoo_stub(
-    stub: str, valid_params: Union[List[str], None] = None
-) -> Tuple[str, Dict[str, str]]:
+class Zoo:
     """
-    :param stub: A SparseZoo model stub. i.e. 'model/stub/path',
-        'zoo:model/stub/path', 'zoo:model/stub/path?param1=value1&param2=value2'
-    :param valid_params: list of expected parameter names to be encoded in the
-        stub. Will raise a warning if any unexpected param names are given. Leave
-        as None to not raise any warnings. Default is None
-    :return: the parsed base stub and a dictionary of parameter names and their values
+    Provides static functions for loading and searching SparseZoo models and recipes
     """
-    # strip optional zoo stub prefix
-    if stub.startswith(ZOO_STUB_PREFIX):
-        stub = stub[len(ZOO_STUB_PREFIX) :]
-
-    if "?" not in stub:
-        return stub, {}
-
-    stub_parts = stub.split("?")
-    if len(stub_parts) > 2:
-        raise ValueError(
-            "Invalid SparseZoo stub, query string must be preceded by only one '?'"
-            f"given {stub}"
-        )
-    stub, params = stub_parts
-    params = dict(param.split("=") for param in params.split("&"))
 
-    if valid_params is not None and any(param not in valid_params for param in params):
-        warnings.warn(
-            f"Invalid query string for stub {stub} valid params include {valid_params},"
-            f" given {list(params.keys())}"
-        )
+    _CONSTRUCTORS = {}
 
-    return stub, params
+    @staticmethod
+    def constructor(construct_key: str):
```

Comment:
@kevinaer do we still need the constructor and register functions now?

Reply:
Ah good catch, we do not. Will get rid of them